File: Numeros_Primos_Java/app/src/main/java/com/example/numeros_primos_java/MainActivity.java
package com.example.numeros_primos_java;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private Button myButton = null;
    private EditText myEditTextNumber = null;
    private TextView myTextView = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        myButton = (Button) findViewById(R.id.button);
        myButton.setOnClickListener(this);
        myEditTextNumber = (EditText) findViewById(R.id.editTextNumber);
        myTextView = (TextView) findViewById(R.id.TextView);

    }

    @Override
    public void onClick(View view) {
        if (myButton.getText().equals(getString(R.string.reiniciar))) {
            myEditTextNumber.setText("");
            myTextView.setText(R.string.introducir);
            myButton.setText(R.string.comprobar);
        }
        else if (myButton.getText().equals(getString(R.string.comprobar))) {
            String inputText = myEditTextNumber.getText().toString();
            if (!inputText.isEmpty()) {
                try {
                    int number = Integer.parseInt(inputText);

                    if (isPrime(number)) {
                        myTextView.setText(R.string.si_primo);
                    } else {
                        myTextView.setText(R.string.no_primo);
                    }

                    myButton.setText(R.string.reiniciar);

                } catch (NumberFormatException e) {
                    myTextView.setText(R.string.error_numero);
                }
            } else {
                myTextView.setText(R.string.error_numero);
            }
        }
    }

    private boolean isPrime(int number) {
        if (number <= 1) {
            return false;
        } else {
            for (int i = 2; i <= number / 2; i++) {
                if (number % i == 0) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: Numeros_Primos_Kotlin/app/src/main/java/com/example/numeros_primos_kotlin/MainActivity.kt
package com.example.numeros_primos_kotlin

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import java.lang.NumberFormatException

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        val myBoton = findViewById<android.widget.Button>(R.id.button)
        val myTextView = findViewById<android.widget.TextView>(R.id.TextView)
        val myEditText = findViewById<android.widget.EditText>(R.id.editTextNumber)

        myBoton.setOnClickListener {

            if (myBoton.text.equals(getText(R.string.reiniciar))) {
                myBoton.text = getText(R.string.comprobar)
                myEditText.setText("")
                myTextView.text = getText(R.string.insertar)

            } else if (myBoton.text.equals(getText(R.string.comprobar))) {
                var numero = myEditText.text.toString().toInt()
                if (myEditText.text.toString().isNotEmpty()) {

                    try {
                        if (esPrimo(numero)) {
                            myTextView.text = getText(R.string.es_primo)
                        } else {
                            myTextView.text = getText(R.string.no_es_primo)
                        }

                        myBoton.text = getText(R.string.reiniciar)
                    }catch (e: NumberFormatException){
                        myTextView.text = getText(R.string.error)
                    }
                }else {
                    myTextView.text = getText(R.string.error)
                }
            }
        }
    }

    private fun esPrimo(numero: Int): Boolean {
        var contador = 2
        var primo = true
        while (primo && contador != numero) {
            if (numero % contador == 0)
                primo = false
            contador++
        }
        return primo
    }

}